.azure-pipelines/generation-templates/capture-openapi.yml

```diff
@@ -13,55 +13,87 @@ parameters:
   default: $(System.ArtifactsDirectory)
 - name: 'cleanMetadataFolder'
   type: string
-steps:
 
-# We only need the scripts
-- checkout: self
-  displayName: checkout generator
-  fetchDepth: 1
-  persistCredentials: true
+jobs:
+- job: get_conversion_settings
+  displayName: "Get conversion settings"
+  steps:
+  - pwsh: |
+      $dirPath = "./conversion-settings"
+      $endpoint = "${{ parameters.endpoint }}"
+      Write-Host "endpoint: $endpoint"
+      Write-Host "directory path: $dirPath"
+      $jsonBase = @{}
+      foreach ($f in Get-ChildItem $dirPath -Recurse -Include *.json) {
+        $name = $f.Name.replace('.json','')
+        $data = @{"File"=$f.Name;"Name"=$name;}
+        $key = "$endpoint-$name"
+        $jsonBase.Add($key,$data)
+      }
+      $json = $jsonBase | ConvertTo-Json -Compress
+      Write-Host "##vso[task.setvariable variable=targets;isOutput=true]$json"
+    name: setTargets
+    workingDirectory: $(Build.SourcesDirectory)/msgraph-metadata
```

Comment:
```yml
- template: checkout-metadata.yml
```
Working directory does not exist yet. Line 56 will need to be called before this.